File: src/MPI_Datatype.c
#include "mpiRC.h"

/* Translate from R int into MPI_Type -
 *
 * Ordering must match MPI_Type.R
 * Null return should never get reached.
*/

MPI_Datatype mpirc_MPI_Datatype(int r_mpi_type) {
  if (r_mpi_type == 1) return MPI_INT;
  if (r_mpi_type == 2) return MPI_DOUBLE;
  return MPI_DATATYPE_NULL;
}

File: R/MPI_Datatype.R
mpirc_MPI_Datatype <- function(type)  {
  type <- toupper(type)

  # Order of Ops (1..) must match numbering in MPI_Type.c

  mpi_Datatype_enum <- c("INT", "DOUBLE")
  stopifnot(type %in% mpi_Datatype_enum)
  as.integer(which(type == mpi_Datatype_enum))
}
